packages/protocol-kit/src/SafeProvider.ts

```diff
@@ -32,49 +22,114 @@ import {
   SafeProviderConfig,
   Eip1193Provider,
   HttpTransport,
-  SocketTransport
+  SocketTransport,
+  ExternalSigner
 } from '@safe-global/protocol-kit/types'
+import { asAddress, asHash, asHex, getChainById } from './utils/types'
+import {
+  createPublicClient,
+  createWalletClient,
+  PublicClient,
+  custom,
+  http,
+  getAddress,
+  isAddress,
+  Transaction,
+  decodeAbiParameters,
+  encodeAbiParameters,
+  parseAbiParameters,
+  toBytes,
+  BlockTag,
+  Chain
+} from 'viem'
+import { privateKeyToAccount } from 'viem/accounts'
+import {
+  toEstimateGasParameters,
+  toCallGasParameters,
+  sameString
+} from '@safe-global/protocol-kit/utils'
+
+function asBlockId(blockId: number | string | undefined) {
+  return typeof blockId === 'number' ? blockNumber(blockId) : blockTag(blockId)
+}
+
+function blockNumber(blockNumber: any) {
+  return { blockNumber: blockNumber.toNumber() }
+}
+
+function blockTag(blockTag: any) {
+  return { blockTag: blockTag as BlockTag }
+}
 
 class SafeProvider {
-  #externalProvider: BrowserProvider | JsonRpcProvider
+  #chain?: Chain
+  #externalProvider: PublicClient
```

Comment:
It could be a WalletClient as well right? If you want to sign or send something using the client i think with the PublicClient you cant

Reply:
Because this was a bigger change, I addressed it in a new PR: #928